core/services/job/spawner.go

```diff
@@ -176,27 +176,28 @@ func (js *spawner) StartService(ctx context.Context, jb Job) error {
 		jb.PipelineSpec.GasLimit = &jb.GasLimit.Uint32
 	}
 
-	services, err := delegate.ServicesForSpec(jb)
+	srvs, err := delegate.ServicesForSpec(jb)
 	if err != nil {
 		js.lggr.Errorw("Error creating services for job", "jobID", jb.ID, "error", err)
 		cctx, cancel := utils.ContextFromChan(js.chStop)
 		defer cancel()
 		js.orm.TryRecordError(jb.ID, err.Error(), pg.WithParentCtx(cctx))
 		js.activeJobs[jb.ID] = aj
-		return nil
+		return errors.Wrapf(err, "failed to create services for job: %d", jb.ID)
 	}
 
-	js.lggr.Debugw("JobSpawner: Starting services for job", "jobID", jb.ID, "count", len(services))
+	js.lggr.Debugw("JobSpawner: Starting services for job", "jobID", jb.ID, "count", len(srvs))
 
-	for _, service := range services {
-		err = service.Start(ctx)
+	var ms services.MultiStart
+	for _, srv := range srvs {
+		err = ms.Start(ctx, srv)
 		if err != nil {
 			js.lggr.Criticalw("Error starting service for job", "jobID", jb.ID, "error", err)
-			continue
```

Comment:
need to either add it to the active jobs map here before returning or remove the active jobs check in delete. Otherwise we will not be able to delete jobs that failed to start

Reply:
I think I prefer leaving it out of active and loosening the delete check... :thinking: WDYT?
I'll push changes in a bit once the tests are fixed up.

Reply:
Oh, but you left a note here: https://github.com/smartcontractkit/chainlink/blob/fe198da564fc6e9a35d753e7242cb7bdbbbdee35/core/services/job/spawner.go#L170-L173

Reply:
It seems like we could just look it up in the case that it is not active though :shrug:

Reply:
I think I've got this working now - just patching up tests that now require some extra dummy config for a peerID/p2pkey so the job creation doesn't fail outright, and some modified checks to permit partial failures (e.g. `contract not deployed at address`) and work around.

Reply:
Greenzo! :green_circle: